### PR TITLE
feat: Implement Request class and tests

### PR DESCRIPTION
**Pull request by ahume**
Hey, @phuu stole my Body commit. Bad OSS manners :-1:

### CODE REVIEW

File: src/.d.ts [hunk 1]
```diff
@@ -31,6 +48,9 @@ declare interface Fetch {
   (request: FetchRequest): Promise<FetchResponse>;
 }
 
+declare type FetchHeaderInit = FetchHeaders|Array<string>
```
Comment (phuu) [inline src/.d.ts]: Semi. Does this lint?

Comment (ahume) [inline src/.d.ts]: Fixed. Yeah, it does. Weird?

File: src/Request/index.test.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    return newRequest.text().then((v) => {
+      t.same(v, "The body");
+    });
+  }
+);
+
+test(
+  "Can create new Request with url string and full init",
+  (t: TestAssertions) => {
+    let request = new Request("mine.json", {
```
Comment (phuu) [inline src/Request/index.test.ts]: We should test `new Request(oldRequest, { some: 'init' })` too.

Comment (ahume) [inline src/Request/index.test.ts]: Fixed.

**Pull request merged**